zkevm-circuits/src/tx_circuit/sign_verify.rs

```diff
@@ -400,139 +377,181 @@ impl<F: Field> SignVerifyChip<F> {
+            .collect_vec();
 
-            let (_, _, msg_hash_rlc) = flex_gate_chip.inner_product(
-                ctx,
-                &assigned_msg_hash_le_selected
-                    .iter()
-                    .map(|x| QuantumCell::Existing(x))
-                    .take(32)
-                    .collect_vec(),
-                &evm_challenge_powers,
-            )?;
-            msg_hash_rlc
+        let pk_y_le = sign_data
+            .pk
+            .y
+            .to_bytes()
+            .iter()
+            .map(|&x| QuantumCell::Witness(Value::known(F::from_u128(x as u128))))
+            .collect_vec();
+        let pk_assigned = ecc_chip.load_private(
+            ctx,
+            (Value::known(sign_data.pk.x), Value::known(sign_data.pk.y)),
+        )?;
+
+        self.assert_crt_int_byte_repr(
+            ctx,
+            &ecdsa_chip.range,
+            &pk_assigned.x,
+            &pk_x_le,
+            &powers_of_256_cells,
+            &None,
+        )?;
+
+        self.assert_crt_int_byte_repr(
+            ctx,
+            &ecdsa_chip.range,
+            &pk_assigned.y,
+            &pk_y_le,
+            &powers_of_256_cells,
+            &None,
+        )?;
+
+        let assigned_pk_le_selected = [pk_y_le, pk_x_le].concat();
+
+        Ok(SignDataDecomposed {
+            pk_hash_cells,
+            msg_hash_cells: assigned_msg_hash_le,
+            pk_cells: assigned_pk_le_selected,
+            address,
+            is_address_zero,
+        })
+    }
+
+    #[allow(clippy::too_many_arguments)]
+    fn halo2_assign_sig_verify(
+        &self,
+        ctx: &mut Context<F>,
+        rlc_chip: &RangeConfig<F>,
+        sign_data: Option<&SignData>,
+        sign_data_decomposed: &SignDataDecomposed<F>,
+        challenges: &Challenges<Value<F>>,
+        sig_is_valid: &AssignedValue<F>,
+    ) -> Result<([AssignedValue<F>; 3], AssignedSignatureVerify<F>), Error> {
+        let (_padding, sign_data) = match sign_data {
+            Some(sign_data) => (false, sign_data.clone()),
+            None => (true, SignData::default()),
         };
-        // println!("halo2ecc assigned msg hash rlc: {:?}", msg_hash_rlc.value());
 
         // ================================================
-        // step 3 random linear combination of pk
+        // step 0. powers of aux parameters
         // ================================================
-        let pk_rlc = {
-            // build ecc chip from Fp chip
-            let ecc_chip = EccChip::<F, FpChip<F>>::construct(ecdsa_chip);
-
-            let pk_x_le = sign_data
-                .pk
-                .x
-                .to_bytes()
-                .iter()
-                .map(|&x| QuantumCell::Witness(Value::known(F::from_u128(x as u128))))
-                .collect_vec();
+        let evm_challenge_powers = iter::successors(Some(F::one()), |coeff| {
+            Some(challenges.evm_word().inner.unwrap() * *coeff)
```

Comment:
nono you have to pick my changes. we should never "challenges.evm_word().inner.unwrap()" .